components/shared/Modal.tsx

```diff
@@ -0,0 +1,266 @@
+import React, { useMemo, useRef, useState } from "react";
+import { Button, Input } from "./index";
+import Select from "./Select";
+import { useAsset, useCreateAsset } from "@livepeer/react";
+import { toast } from "react-hot-toast";
+import Link from "next/link";
+
+interface ModalProps {
+  onClose: () => void;
+}
+
+const Modal = ({ onClose }: ModalProps) => {
+  const [chain, setChain] = useState<string>("ethereum");
+  const [tokenAddress, setTokenAddress] = useState<string>();
+  const [tokenAmount, setTokenAmount] = useState<string>();
+  const [userAddress, setWalletAddress] = useState<string>();
+  const [file, setFile] = useState<File>();
+
+  const fileInputRef = useRef<HTMLInputElement>(null);
+
+  const {
+    mutate: createAsset,
+    data: createdAsset,
+    status: createStatus,
+    progress,
+  } = useCreateAsset(
+    file
+      ? {
+          sources: [
+            {
+              file: file,
+              name: file.name,
+              playbackPolicy: {
+                type: "webhook",
+                webhookId: process.env.NEXT_PUBLIC_WEBHOOK_ID as string,
+                webhookContext: {
+                  requirement: {
+                    token: {
+                      chain,
+                      tokenAddress,
+                      tokenAmount,
+                    },
+                    userAddress,
+                  },
+                },
+              },
+            },
+          ] as const,
+        }
+      : null
+  );
+
+  const { data: asset, status: assetStatus } = useAsset({
+    assetId: createdAsset?.[0]?.id,
+    refetchInterval: (asset) =>
+      asset?.storage?.status?.phase !== "ready" ? 2000 : false,
+  });
+
+  const progressFormatted = useMemo(() => {
+    if (progress?.[0]?.phase === "failed" || createStatus === "error") {
+      return "Failed to upload video.";
+    } else if (progress?.[0]?.phase === "waiting") {
+      return "Waiting";
+    } else if (progress?.[0]?.phase === "uploading") {
+      return `Uploading: ${Math.round(progress?.[0]?.progress * 100)}%`;
+    } else if (progress?.[0]?.phase === "processing") {
+      return `Processing: ${Math.round(progress?.[0]?.progress * 100)}%`;
+    }
+    return null;
+  }, [progress, createStatus]);
+
+  const isLoading = useMemo(() => {
+    return (
+      createStatus === "loading" ||
+      assetStatus === "loading" ||
+      (asset && asset?.status?.phase !== "ready") ||
+      (asset?.storage && asset?.storage?.status?.phase !== "ready")
+    );
+  }, [asset, assetStatus, createStatus]);
+
+  const validateInputs = () => {
+    const userAddressRegex = /^0x[a-fA-F0-9]{40}$/;
+    const tokenAddressRegex = /^0x[a-fA-F0-9]{40}$/;
```

Comment:
these are not actual valid to check if addresses are valid, you can use utils.isAddress from ethers